src/dx_azure_iot.c

```diff
@@ -442,10 +449,37 @@ static bool ConnectToIotHub(const char *hostname)
         return false;
     }
 
-    if ((iothubClientHandle = IoTHubDeviceClient_LL_CreateWithAzureSphereFromDeviceAuth(hostname, &MQTT_Protocol)) == NULL) {
-        Log_Debug("ERROR: Failed to create client IoT Hub Client Handle\n");
-        return false;
+    if((_proxyConfig->proxyEnabled) && (_proxyConfig->proxyAddress != NULL)){
+
+        iothubClientHandle =
```

Comment:
Could this be moved to a function - ideally in a iot_proxy.c/.h file - would make it easier to understand what is going on and maintain